XiEditor/SidebarItems.swift

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Singleton used across window sidebars containing an array of file system items
+/// Provides simple methods to modify the items
+/// Can listen to changes in the items by hooking into the "sidebarItemsChanged" notification
+final class SidebarItems {
+
+    static let sharedInstance = SidebarItems()
+    public var items: [FileSystemItem]
```

Comment:
Perhaps this can be `public var items = [FileSystemItem]()`, and you can remove the init that's below it for more clarity.